tests/nftables/test_nftables.py

```diff
@@ -0,0 +1,191 @@
+
+        # IPV4
+        output = json.loads(run_cmd(["nft", "-j", "list", "table", "ip", "crowdsec"]))
+        sets = {
+            (node["set"]["family"], node["set"]["name"], node["set"]["type"])
+            for node in output["nftables"]
+            if "set" in node
+        }
+        assert ("ip", "crowdsec-blacklists", "ipv4_addr") in sets
+        rules = {
+            node["rule"]["chain"] for node in output["nftables"] if "rule" in node
+        }  # maybe stricter check ?
+        assert "crowdsec-chain" in rules
+
+        # IPV6
+        output = json.loads(run_cmd(["nft", "-j", "list", "table", "ip6", "crowdsec6"]))
+        sets = {
+            (node["set"]["family"], node["set"]["name"], node["set"]["type"])
+            for node in output["nftables"]
+            if "set" in node
+        }
+        assert ("ip6", "crowdsec6-blacklists", "ipv6_addr") in sets
+
+        rules = {
+            node["rule"]["chain"] for node in output["nftables"] if "rule" in node
+        }  # maybe stricter check ?
+        assert "crowdsec6-chain" in rules
+
+    def test_duplicate_decisions_across_decision_stream(self):
+        d1, d2, d3 = generate_n_decisions(3, dup_count=1)
+        mock_lapi.ds.insert_decisions([d1])
+        sleep(1)
+        self.assertEqual(
+            self.get_set_elements("ip", "crowdsec", "crowdsec-blacklists"), {"0.0.0.0"}
+        )
+
+        mock_lapi.ds.insert_decisions([d2, d3])
+        sleep(1)
+        assert self.fb.poll() is None
+        self.assertEqual(
+            self.get_set_elements("ip", "crowdsec", "crowdsec-blacklists"), {"0.0.0.0", "0.0.0.1"}
+        )
+
+        mock_lapi.ds.delete_decision_by_id(d1["id"])
+        mock_lapi.ds.delete_decision_by_id(d2["id"])
+        sleep(1)
+        self.assertEqual(self.get_set_elements("ip", "crowdsec", "crowdsec-blacklists"), set())
+        assert self.fb.poll() is None
+
+        mock_lapi.ds.delete_decision_by_id(d3["id"])
+        sleep(1)
+        self.assertEqual(self.get_set_elements("ip", "crowdsec", "crowdsec-blacklists"), set())
+        assert self.fb.poll() is None
+
+    def test_decision_insertion_deletion_ipv4(self):
+        total_decisions, duplicate_decisions = 100, 23
+        decisions = generate_n_decisions(total_decisions, dup_count=duplicate_decisions)
+        mock_lapi.ds.insert_decisions(decisions)
+        sleep(1)  # let the bouncer insert the decisions
+
+        set_elements = self.get_set_elements("ip", "crowdsec", "crowdsec-blacklists")
+        self.assertEqual(len(set_elements), total_decisions - duplicate_decisions)
+        assert set([i["value"] for i in decisions]) == set_elements
+        assert "0.0.0.0" in set_elements
+
+        mock_lapi.ds.delete_decisions_by_ip("0.0.0.0")
+        sleep(1)
+
+        set_elements = self.get_set_elements("ip", "crowdsec", "crowdsec-blacklists")
+        assert set([i["value"] for i in decisions if i["value"] != "0.0.0.0"]) == set_elements
```

Comment:
python has set comprehension: set(i["value"] ... "0.0.0.0"), no need to create a list. Or literal: {i["value"].... "0.0.0.0"}